tinygrad/codegen/linearizer.py

```diff
@@ -513,28 +523,35 @@ def uop(self, uop:UOps, dtype:Optional[DType]=None, vin:Tuple[UOp, ...]=tuple(),
     if cachable: self.saved_exprs[key] = ret
     return ret
 
+  def parse_cast(self, x, u):
+    dtype,bitcast = x.arg
+    if not isinstance(dtype, ImageDType): return self.uop(UOps.CAST, dtype, (u,), (dtype,bitcast))
+    return u if u.dtype == dtype.base else self.uop(UOps.CAST, dtype.base, (u,), (dtype.base,False)) # we don't bitcast on images
```

Comment:
are we bitcasting on image now?

Reply:
no the second arg is False

Reply:
if we are not bitcasting now, can it be `self.uop(UOps.CAST, dtype, (u,), (dtype,bitcast))` and merge with the line above?